modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/model/DeclaringPointElement.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.introspection.describer.model;
+
+/**
+ * A generic contract for any component that is considered as declaring point of different extensions components (Sources,
+ * Connection Providers and Operation Containers)
+ * <p>
+ * The ones considered as declaring points are the Extensions and Extension's configurations.
+ *
+ * @since 4.0
+ */
+public interface DeclaringPointElement extends WithMessageSources, WithOperationContainers, WithConnectionProviders, ParameterizableTypeElement
```

Comment:
DeclaringPOintElement -> ComponentElement....

Same for the wrapper -> ComponentElementWrapper